core/src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -235,6 +235,23 @@ object ConfigConvert extends LowPriorityConfigConvertImplicits {
           }
 
           tryBuilder.map(_.result())
+        case o: ConfigObject =>
```

Comment:
I think it's a good idea to extract this as a function. I will incorporate it in to the PR. As for the change with ```Throwable```. The failure is caught with ```NonFatal``` in ```Try(keyString.toInt -> value)``` anyways. But it is probably good style to still match on it again.

As for the signature of the method. Should we change it to the following to not be using ```Nothing``` as type?
```private def reportBadIndex[U](keyString: String): PartialFunction[Throwable, Failure[U]]```